pkg/controller/certificates/sync.go

```diff
@@ -104,6 +107,30 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (err e
 		return nil
 	}
 
+	// check that certificate secret name is unique
+	namespaceCrts := c.certificateLister.Certificates(crtCopy.Namespace)
+	otherCrts, err := namespaceCrts.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	hasDuplicate := false
+	for _, otherCrt := range otherCrts {
+		if otherCrt.Name != crtCopy.Name && otherCrt.Spec.SecretName == crtCopy.Spec.SecretName {
+			hasDuplicate = true
+			break
+		}
+	}
+	if hasDuplicate {
+		c.Recorder.Eventf(crtCopy, corev1.EventTypeWarning, errorDuplicateSecretName, "Duplicate secretName %v", crtCopy.Spec.SecretName)
```

Comment:
Perhaps extend this log message to also advise users how to fix the issue? `Another certificate already specifies the same spec.secretName, please update the secret name on either certificate` or something to that effect 😄